pkg/git/options.go

```diff
@@ -72,6 +72,16 @@ type AuthOptions struct {
 	Identity   []byte
 	KnownHosts []byte
 	CAFile     []byte
+	// TransportOptionsURL is a unique identifier for this set of authentication
+	// options. It's used by managed libgit2 transports to uniquely identify
+	// which credentials to use for a particular Git operation, and avoid misuse
+	// of credentials in a multi-tenant environment.
+	// It must be prefixed with a valid transport protocol ("ssh:// "or "http://") because
+	// of the way managed transports are registered and invoked.
+	// It's a field of AuthOptions despite not providing any kind of authentication
+	// info, as it's the only way to sneak it into git.Checkout, without polluting
+	// it's args and keeping it generic.
+	TransportOptionsURL string
```

Comment:
This leaks implementation details into the generic options and should be configured via other means.

Reply:
Added this to the issue tracking the consolidation of the git implementation.